### PR TITLE
Added message for imported tests count and refactored TestsController to use only services

### PR DESCRIPTION
**Pull request by gogo4ds**
Add info message when test output is empty and answer is wrong
Closes https://github.com/SoftUni-Internal/suls-issues/issues/4459

### CODE REVIEW

File: Open Judge System/OJS.Services/Business/Submissions/SubmissionsBusinessService.cs [hunk 1]
```diff
@@ -49,6 +52,89 @@ public IQueryable<Submission> GetAllForArchiving()
...
+                    .Include(s => s.TestRuns)
+                    .Include(s => s.TestRuns.Select(tr => tr.Test))
+                    .ToList();
+
+                var submissionResults = problemSubmissions
+                    .Select(s => new
+                    {
+                        s.Id,
+                        s.ParticipantId,
+                        CorrectTestRuns = s.TestRuns.Count(t => t.ResultType == TestRunResultType.CorrectAnswer && !t.Test.IsTrialTest),
```
Comment (achobanov) [inline Open Judge System/OJS.Services/Business/Submissions/SubmissionsBusinessService.cs]: Isn't this line a bit too long?

File: Open Judge System/OJS.Services/Business/Submissions/SubmissionsBusinessService.cs [hunk 2]
```diff
@@ -49,6 +52,89 @@ public IQueryable<Submission> GetAllForArchiving()
...
+                var problemSubmissionsById = problemSubmissions.ToDictionary(s => s.Id);
+                var topResults = new Dictionary<int, ParticipantScoreModel>();
+
+                foreach (var submissionResult in submissionResults)
+                {
+                    var submission = problemSubmissionsById[submissionResult.Id];
+                    var points = 0;
+                    if (submissionResult.AllTestRuns != 0)
+                    {
+                        points = (submissionResult.CorrectTestRuns * submissionResult.MaxPoints) / submissionResult.AllTestRuns;
```
Comment (achobanov) [inline Open Judge System/OJS.Services/Business/Submissions/SubmissionsBusinessService.cs]: Isn't this line a bit too long?

File: Open Judge System/OJS.Services/Data/Submissions/SubmissionsDataService.cs [hunk 3]
```diff
@@ -75,5 +75,12 @@ public void Update(Submission submission)
             this.submissions.Update(submission);
             this.submissions.SaveChanges();
         }
+
+        public void RemoveTestRunsCacheByProblem(int problemId) =>
+            this.GetAllByProblem(problemId)
+            .Update(s => new Submission
```
Comment (achobanov) [inline Open Judge System/OJS.Services/Data/Submissions/SubmissionsDataService.cs]: Tab

File: Open Judge System/Web/OJS.Web.Common/ZippedTestManipulator/ZippedTestsParser.cs [hunk 4]
```diff
@@ -50,44 +52,49 @@ public static void AddTestsToProblem(Problem problem, TestsParseResult tests)
                 });
 
                 zeroTestsOrder++;
+                addedTestsCount++;
             }
 
             var lastTest = problem.Tests.Where(x => !x.IsTrialTest).OrderByDescending(x => x.OrderBy).FirstOrDefault();
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web.Common/ZippedTestManipulator/ZippedTestsParser.cs]: This line seems a bit long.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/KendoRemoteDataController.cs [hunk 5]
```diff
@@ -54,6 +58,29 @@ public JsonResult GetUsersContaining(string userFilter)
...
+            if (!string.IsNullOrWhiteSpace(text))
+            {
+                problems = this.problemsData
+                    .GetAll()
+                    .Where(p => p.Name.Contains(text))
+                    .Take(DefaultItemsToTake);
+            }
+
+            var result = problems
+                .Select(p => new DropdownViewModel
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/KendoRemoteDataController.cs]: I personally like this formatting, but I think `Select` should be on the above like. Similarly to `ZippedTestsParser`, line 84.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/KendoRemoteDataController.cs [hunk 6]
```diff
@@ -98,5 +125,20 @@ public JsonResult GetCascadeProblemGroupsFromContest(int contestId)
...
             return this.Json(result, JsonRequestBehavior.AllowGet);
         }
+
+        [AjaxOnly]
+        public JsonResult GetCascadeProblemsFromContest(int contestId)
+        {
+            var problems = this.problemsData.GetAllByContest(contestId);
+
+            var result = problems
+                .Select(p => new DropdownViewModel
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/KendoRemoteDataController.cs]: I personally like this formatting, but I think `Select` should be on the above like. Similarly to `ZippedTestsParser`, line 84.
